src/regex.c

```diff
@@ -286,19 +287,22 @@ int do_regex_hybrid_crack(struct db_main *db, const char *regex,
 	}
 
 	while (c_iterator_next(iter)) {
-		c_iterator_value(iter, buffer);
-		c_simplestring_to_utf8_string(buffer, &word[0], sizeof(word));
 		c_simplestring_clear(buffer);
-		if (options.internal_cp != UTF_8)
-			utf8_to_cp_r(word, word, sizeof(word));
+		c_iterator_value(iter, buffer);
+		word = c_simplestring_to_string(buffer);
+    /**
+      * rexgen already creates the correct encoding
+      */
+		//if (options.internal_cp != UTF_8)
+		//	utf8_to_cp_r(word, word, sizeof(word));
 		if (options.mask) {
```

Comment:
Here's some of the locale caveats we're facing in JtR

The **terminal** may be UTF-8 or something else (but nowadays it's nearly always UTF-8). Independant of terminal, we have an **input encoding** (hash files, wordlists) and a **target encoding** (whatever encoding was used for the hashes we're attacking - eg. for LM hashes it's often CP850). Finally, for the case of UTF-8 candidates running through eg. permutation rules, we have the concept of an **internal encoding** which is much easier to process for case switching etc. So even when everything else is UTF-8, we often use CP1252 as internal encoding.